src/metabase/driver/redshift.clj

```diff
@@ -0,0 +1,97 @@
+                       JOIN pg_attribute source_column ON c.conrelid     = source_column.attrelid
+                       JOIN pg_class dest_table        ON c.confrelid    = dest_table.oid
+                       JOIN pg_attribute dest_column   ON c.confrelid    = dest_column.attrelid
+                     WHERE c.contype            = 'f'::char
+                       AND source_table.relname = ?
+                       AND n.nspname            = ?
+                       AND source_column.attnum = ANY(c.conkey)
+                       AND dest_column.attnum   = ANY(c.confkey)"
+                    (:name table)
+                    (:schema table)])))
+
+(defrecord RedshiftDriver []
+  clojure.lang.Named
+  (getName [_] "Amazon Redshift"))
+
+(extend RedshiftDriver
+  driver/IDriver
+  (merge (sql/IDriverSQLDefaultsMixin)
+         {:date-interval  date-interval
+          :details-fields (constantly [{:name         "host"
+                                        :display-name "Host"
+                                        :placeholder  "my-cluster-name.abcd1234.us-east-1.redshift.amazonaws.com"
+                                        :required     true}
+                                       {:name         "port"
+                                        :display-name "Port"
+                                        :type         :integer
+                                        :default      5439}
+                                       {:name         "db"
+                                        :display-name "Database name"
+                                        :placeholder  "toucan_sightings"
+                                        :required     true}
+                                       {:name         "user"
+                                        :display-name "Master username"
+                                        :placeholder  "cam"
+                                        :required     true}
+                                       {:name         "password"
+                                        :display-name "Master user password"
+                                        :type         :password
+                                        :placeholder  "*******"
+                                        :required     true}])
+          :table-fks      table-fks})
+
+  sql/ISQLDriver
+  (merge postgres/PostgresISQLDriverMixin
+         {:connection-details->spec  connection-details->spec
+          :current-datetime-fn       (constantly (k/sqlfn* :GETDATE))
+          :set-timezone-sql          (constantly nil)
+          :unix-timestamp->timestamp unix-timestamp->timestamp}
+         ;; HACK ! When we test against Redshift we use a session-unique schema so we can run simultaneous tests against a single remote host;
+         ;; when running tests tell the sync process to ignore all the other schemas
+         (when config/is-test?
+           {:excluded-schemas (memoize
```

Comment:
we do this kind of stuff a few other places in the code (`config/is-test?`), which is a shame, because it would really be nicer if unit testing specific conditions were kept in the test code.

i'm just curious as i look at this if it's possible in this case to put something like this in the test driver interface or somewhere in the test initialization code such that we take the normal driver that is registered, modify it to overload a function like this, and then re-register the modified version.

it's not a big deal right now, but as the codebase grows i think we'll want to be more diligent about finding ways to prevent code like this from being part of the main app codebase since it's specific to the unit testing environment.